columnar/src/backend/columnar/columnar_tableam.c

```diff
@@ -958,13 +1007,82 @@ columnar_tuple_update(Relation relation, ItemPointer otid, TupleTableSlot *slot,
 }
 
 
+static TM_Result
+columnar_tuple_satisfies_update(HeapTuple htup, CommandId curcid,
+						Buffer buffer)
+{
+	HeapTupleHeader tuple = htup->t_data;
+
+	Assert(ItemPointerIsValid(&htup->t_self));
+	Assert(htup->t_tableOid != InvalidOid);
+#if 0
+	if (!HeapTupleHeaderXminCommitted(tuple))
+	{
+		if (HeapTupleHeaderXminInvalid(tuple))
+			return TM_Invisible;
+
+		/* Used by pre-9.0 binary upgrades */
+		if (tuple->t_infomask & HEAP_MOVED_OFF)
+		{
+			TransactionId xvac = HeapTupleHeaderGetXvac(tuple);
+
+			if (TransactionIdIsCurrentTransactionId(xvac))
+				return TM_Invisible;
+			if (!TransactionIdIsInProgress(xvac))
+			{
+				if (TransactionIdDidCommit(xvac))
+				{
+					return TM_Invisible;
+				}
+			}
+		}
+
+	}
+#endif
+	return TM_Ok;
+}
+
+
+
 static TM_Result
 columnar_tuple_lock(Relation relation, ItemPointer tid, Snapshot snapshot,
 					TupleTableSlot *slot, CommandId cid, LockTupleMode mode,
 					LockWaitPolicy wait_policy, uint8 flags,
 					TM_FailureData *tmfd)
 {
-	elog(ERROR, "columnar_tuple_lock not implemented");
+	// copy/paste from columnar_fetch_row_version
```

Comment:
Please recheck this function, remove redundant comments and check memory context that is used here. This was supposed to be PoC implementation.

Reply:
found a couple of small issues with the function and fixed.